OpenSim/Simulation/Model/Model.h

```diff
@@ -1021,8 +1037,10 @@ OpenSim_DECLARE_CONCRETE_OBJECT(Model, ModelComponent);
     // _workingState will be set to the system default state when
     // initializeState() or initSystem() is called.
     SimTK::State _workingState;
-
-
+    
+    //Local reference of the State's controlsCache to avoid the cost of fetching
+    //the unchanging cache from the State
+    SimTK::Measure_<SimTK::Vector>::Result _controlsCache;
```

Comment:
This is redundant with the member variable `_modelControlsIndex`. Only one or the other should be stored. We discussed which one should be stored in this issue: https://github.com/opensim-org/opensim-core/issues/411.

So if possible can you remove the `_modelControlsIndex` member? This will require some work because `_modelControlsIndex` is used in a few places. I think it might be best to do this in a separate pull request *into* this branch (instead of into master).

As a helper for how to do this, instead of holding onto `_modelControlsIndex` in `extendAddToSystem()`, you'll just hold onto what is currently called `modelControls` instead.